### ISSUE TITLE
refactor setting of use of DHCP for NTP servers

### ISSUE THREAD

**Issue by richm** [linux-system-roles/timesync#93]
https://github.com/linux-system-roles/timesync/pull/92 has the logic in the template to consider various versions of chrony and operating system and platform.  This should be refactored into `vars/` files.

Have `vars/Fedora.yml` like this:
```yaml
timesync_can_use_dhcp_ntp_servers: "{{ 'yes' if __timesync_chrony_version is version('4.0', '>=') else 'no' }}"
```
Then for the fedora platforms that do not support it e.g. `vars/Fedora_32.yml`, `vars/Fedora_31.yml` (don't need to go back before that as Fedora 30 is EOL):
```yaml
timesync_can_use_dhcp_ntp_servers: no
```
create `vars/RedHat.yml` like this:
```yaml
timesync_can_use_dhcp_ntp_servers: no
```
Also add that to `vars/RedHat_6.yml` and `vars/CentOS_6.yml`

Then create `vars/RedHat_9.yml` and `vars/CentOS_9.yml` like Fedora.yml:
```yaml
timesync_can_use_dhcp_ntp_servers: "{{ 'yes' if __timesync_chrony_version is version('4.0', '>=') else 'no' }}"
```

Then the logic in templates/chrony.conf.j2 can be simplified like this:
```

{% if timesync_dhcp_ntp_servers and timesync_can_use_dhcp_ntp_servers %}
# Use NTP servers from DHCP.
sourcedir /run/chrony-dhcp

{% endif %}

**Comment by mlichvar** [comment]
Ok, that makes sense, but I think the variable name should be different. In older Fedora versions the loading of servers from DHCP was implicit. It could be disabled only by adding PEERNTP=no to /etc/sysconfig/network. In Fedora 33 that still works, but it also can be disabled by removing the sourcedir line from chrony.conf.

Maybe this should be included in #92?

**Comment by richm** [comment]
> Ok, that makes sense, but I think the variable name should be different.

ok

> In older Fedora versions the loading of servers from DHCP was implicit. It could be disabled only by adding PEERNTP=no to /etc/sysconfig/network. In Fedora 33 that still works, but it also can be disabled by removing the sourcedir line from chrony.conf.
> 
> Maybe this should be included in #92?

Not sure.  If you think so, ok.

**Comment by mlichvar** [comment]
I've reworked #92 to use this approach, adding new var files to have distro-specific location of the DHCP sourcedir.

Thanks for the suggestions.
